File: go/xortag.go
package main

import (
	"encoding/json"
	"fmt"
	"io"
	"net/http"
	"os"
	"time"
	"math/rand"
)

const baseUrl = "https://xortag.azurewebsites.net/"
var world map[string]interface{} = nil

func main() {
	register()

	for true {
		switch rand.Intn(4) {
		case 0:
			moveUp()
		case 1:
			moveDown()
		case 2:
			moveLeft()
		case 3:
			moveRight()
		}
		time.Sleep(1 * time.Second)
	}
}

func register() {
	world = send(baseUrl + "register")
	fmt.Println("You successfully registered")
	fmt.Println("Your player name is " + world["name"].(string) + " and your id is " +  fmt.Sprintf("%g", world["id"].(float64)))
	time.Sleep(1 * time.Second)
}

func moveUp() {
	world = send(baseUrl + "moveup/" + fmt.Sprintf("%g", world["id"].(float64)))
}

func moveDown() {
	world = send(baseUrl + "movedown/" + fmt.Sprintf("%g", world["id"].(float64)))
}

func moveLeft() {
	world = send(baseUrl + "moveleft/" + fmt.Sprintf("%g", world["id"].(float64)))
}

func moveRight() {
	world = send(baseUrl + "moveright/" + fmt.Sprintf("%g", world["id"].(float64)))
}

func send(url string) map[string]interface{} {
	req, err := http.NewRequest("GET", url, nil)
	if err != nil {
		os.Exit(1)
	}
	
	client := &http.Client{}
	response, err := client.Do(req)
	if err != nil {
		os.Exit(2)
	}

	defer response.Body.Close()

	b, err := io.ReadAll(response.Body)
	if err != nil {
		fmt.Println(err)
		os.Exit(3)
	}

	result := make(map[string]interface{})
	json.Unmarshal([]byte(b), &result)
	return result
}

File: php/xortag.php
<?php

echo 'Hello world';

class XorTag {
    private $baseUrl = 'https://xortag.azurewebsites.net/';
    private $world;

    function register() {
        $result = file_get_contents($this->baseUrl . 'register');
        $deserialized = json_decode($result);
        $this->world = $deserialized;
        echo 'You have successfully registered!' . PHP_EOL;
        echo 'Your player name is ' . $this->world->name . ' and your id is ' . $this->world->id . PHP_EOL;
        sleep(1);
    }

    function moveUp() {
        $this->world = json_decode(file_get_contents($this->baseUrl . 'moveup/' . $this->world->id));
    }

    function moveDown() {
        $this->world = json_decode(file_get_contents($this->baseUrl . 'movedown/' . $this->world->id));
    }

    function moveLeft() {
        $this->world = json_decode(file_get_contents($this->baseUrl . 'moveleft/' . $this->world->id));
    }

    function moveRight() {
        $this->world = json_decode(file_get_contents($this->baseUrl . 'moveright/' . $this->world->id));
    }

    function look() {
        $this->world = json_decode(file_get_contents($this->baseUrl . 'look/' . $this->world->id));
    }

    function play() {
        $this->register();
        while(true) {
            switch (rand(1, 4)) {
                case 1:
                    $this->moveUp();
                    break;
                case 2:
                    $this->moveDown();
                    break;
                case 3:
                    $this->moveLeft();
                    break;
                case 4:
                    $this->moveRight();
                    break;
            }
            sleep(1);
        }
    }
}

$player = new XorTag();
$player->play();